### PR TITLE
Fixing issue with 'constructor' variable name

### PR DESCRIPTION
**Pull request by jguille2**
Fixing problem with "constructor" variable name related to #2525

### CODE REVIEW

**Comment by jmoenig** [comment]
yeah, this would work. What do you think about replacing the current ```vars``` object / dictionary with a ```Map``` structure? That's probably a lot cleaner and might even be a little faster. I'm leaning towards this...

File: src/threads.js [hunk 1]
```diff
@@ -5922,7 +5922,7 @@ VariableFrame.prototype.silentFind = function (name) {
     answer the closest variable frame containing
     the specified variable. Otherwise return null.
 */
-    if (this.vars[name] !== undefined) {
+    if (this.vars[name] !== undefined && !(this.vars[name] instanceof Function)) {
```
Comment (jmoenig) [inline src/threads.js]: I guess the idea is that every entry in the ```vars``` dictionary has to be of type ```Variable```. So, we could easily check that, which might be safer and easier. Let me investigate....

**Comment by jmoenig** [comment]
I've decided to check for ```Variable``` type entries only, which is easier and seems to solve the issue. Thank you, @jguille2, for your excellent-as-always analysis of the underlying issue, you're a fantastic help!

**Comment by jguille2** [comment]
Yes Jens,
Checking for  `Variable` type is clearly the right solution, and also is solving that issue.
I really appreciate your words. For me, it is a joy to be part of this great project.
Joan

**Pull request closed without merging**